src/ansible_navigator/actions/run.py

```diff
@@ -166,6 +168,32 @@ def filter_content_keys(obj: dict[Any, Any]) -> dict[Any, Any]:
     return {k: v for k, v in obj.items() if not (k.startswith("_") or k.endswith("uuid"))}
 
 
+def check_playbook(playbook: str) -> str:
+    """Get the full path to the playbook.
+
+    Check the extension of playbook. Whether it is a playbook with yml/
+    yaml extension or playbook provided by a collection or something else
+    and return the path accordingly.
+
+    :param playbook: Provided playbook path
+    :returns: Full playbook path
+    """
+    my_list = [".yaml", ".yml"]
+    if any(playbook.endswith(s) for s in my_list):
+        playbook_path = playbook
+    else:
+        playbook_path = playbook
+        start_path = playbook.rsplit("/", 1)[0]
+        playbook_name = playbook.split(".")[-1]
+        end_paths = [f"{playbook_name}.yml", f"{playbook_name}.yaml"]
+        for end_path in end_paths:
+            file = glob.glob(os.path.join(start_path, "**", end_path), recursive=True)
+            if len(file) != 0:
+                playbook_path = file[0]
+                break
+    return playbook_path
```

Comment:
I'm not sure this code will work the way users will ultimately want it to.  When calling a playbook from a collection, it is just represented by an FQCN and should not be referenced by path.

Additionally, an FQCN for a playbook does support a special case of including the file extension, but that is more for compatibility than anything.

An FQCN for a playbook is just like anything else:

```
{namespace}.{name}.{resource}
```

So we shouldn't attempt to dereference a FQCN to a file path I don't think.  If we did that, then the playbook would not be treated as part of a collection. ansible-core only knows a playbook is part of a collection if the playbook resource that is passed to `ansible-playbook` is an FQCN.  Otherwise, it's just a path to an arbitrary playbook file, without "self" collection context.

Reply:
To clarify one point, both of these are equivalent for `ansible-playbook`:

```
ansible-playbook namespace.name.some_playbook
ansible-playbook namespace.name.some_playbook.yml
```